src/yunohost/domain.py

```diff
@@ -218,3 +237,57 @@ def domain_remove(auth, domain, force=False):
     hook_callback('post_domain_remove', args=[domain])
 
     msignals.display(m18n.n('domain_deleted'), 'success')
+
+
+def domain_letsencrypt(auth, domain, create=False, renew=False, revoke=False):
+    """
+    Manage let's encrypt certificate for a domain
+
+    Keyword argument:
+        domain -- Domain to delete
+        create -- Create a Let's encrypt certificate
+        renew -- Force renewal of a Let's encrypt certificate
+        revoke -- Revoke a Let's encrypt certificate
+
+    """
+    if domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
+
+    if create and not renew and not revoke:
+
+        # backup self signed certificate if exist
+        if os.path.exists('/etc/yunohost/certs/%s/cert.pem' % domain):
+            os.system('mkdir -p /etc/yunohost/certs/%s/yunohost_self_signed' % domain)
+            os.system('sudo mv /etc/yunohost/certs/%s/*.pem /etc/yunohost/certs/%s/*.cnf /etc/yunohost/certs/%s/yunohost_self_signed/' % domain)
+            os.system('sudo rm -f /etc/yunohost/certs/%s/*.pem /etc/yunohost/certs/%s/*.cnf' % domain)
+
+        # create certificate
+        try:
+            os.system('sudo /root/letsencrypt/letsencrypt-auto -a webroot --renew-by-default --agree-dev-preview --agree-tos --webroot-path /etc/letsencrypt/webrootauth -m root@%s -d %s auth' % domain)
+            # restore right for metronome
+            os.system('sudo chown root:metronome /etc/letsencrypt/archive/%s/*' % domain)
+            # create cron
+            os.system('echo "@monthly root yunohost domain letsencrypt -r %s" > /etc/cron.d/letsencrypt-%s' % domain)
+            # symbolic link for cert and key
+            os.system('ln -s /etc/letsencrypt/live/%s/privkey.pem /etc/yunohost/certs/%s/key.pem' % domain)
+            os.system('ln -s /etc/letsencrypt/live/%s/fullchain.pem /etc/yunohost/certs/%s/crt.pem' % domain)
+            msignals.display(m18n.n('domain_letsencrypt_created'), 'success')
+        except:
```

Comment:
A quick hack in the mean time would be to have a

```python
assert os.system(sutff) == 0
```

But it should remain as a quick hack.